modules/Oculus_module/src/HeadRetargeting.cpp

```diff
@@ -7,30 +7,81 @@
  * @date 2018
  */
 
-#include "HeadRetargeting.hpp"
-#include "Utils.hpp"
+// iDynTree
+#include <iDynTree/Core/EigenHelpers.h>
+#include <iDynTree/yarp/YARPConversions.h>
+#include <iDynTree/yarp/YARPEigenConversions.h>
 
-bool HeadRetargeting::configure(const yarp::os::Searchable& config)
+#include <HeadRetargeting.hpp>
+#include <Utils.hpp>
+
+void HeadRetargeting::inverseKinematics(const iDynTree::Rotation& matrix, double& neckPitch,
+                                        double& neckRoll, double& neckYaw)
+{
+    // YXZ decomposition
```

Comment:
If you have a reference (paper/book/other code) on this code, feel free to add it.